src/ThriftSQL/ThriftStream.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace ThriftSQL;
+
+class ThriftStream implements \Iterator {
+	/**
+	 * @var \ThriftSQL
+	 */
+	private $query;
+	private $buffer;
+	private $location;
+
+	/**
+	 * @var \ThriftSQLQuery
+	 */
+	private $stream;
+	private $queryStr;
+
+	public function __construct( $query, $queryStr ) {
+		$this->query    = $query;
+		$this->queryStr = $queryStr;
+	}
+
+	/**
+	 * Return the current element
+	 * @link http://php.net/manual/en/iterator.current.php
+	 * @return mixed Can return any type.
+	 * @since 5.0.0
+	 */
+	public function current() {
+		return $this->buffer[0];
+	}
+
+	/**
+	 * Move forward to next element
+	 * @link http://php.net/manual/en/iterator.next.php
+	 * @return void Any returned value is ignored.
+	 * @since 5.0.0
+	 */
+	public function next() {
+		$this->location ++;
+		array_pop( $this->buffer );

```

Comment:
Shouldn't this be `array_shift()` so that we grab the first element in the buffer? Otherwise if we get a resultset of 6 rows in two chunks like so:

```
---
1
2
3
---
4
5
6
---
```

Iterating through the set would end up producing (I think):

```
1
1
1
4
4
4
```